### ISSUE TITLE
Benchmark?

### ISSUE THREAD

**Issue by peci1** [altineller/fximu2#2]
Do you plan to make some benchmarks that would support your claims about performance? E.g. putting it inside a mobile robot and doing tens/hundreds of meters of dead reckoning... And comparing the results to some standard IMU, e.g. xsens or similar?

**Comment by altineller** [comment]
Hello @peci1 

I have a small robot, which I developed the IMU for. It can do odometry, and robot_pose_ekf fusion with imu sensor, to generate corrected pose.

I also wrote simple python programs, that makes the robot go 1m, and then go 1m in opposite direction. I will be plotting the trajectory with old version vs new version, and produce outputs. Actually I just sat down to do it now.

I also noticed https://github.com/gaowenliang/imu_utils which offers a ros package tool to analyze the IMU performance. (and compare them with bag files, appearently) and they have bag files for other imu's. 

The imu has still place to go for development, namely the hardware based high pass and low pass filters on the sensors, that provide robustness from things like motor noise, etc. 

I worked with mpu6050, mpu9150, mpu9250, bno055 and especially the DMP featured ones were disappointment. I once worked with a UM6, but I never owned one. It works better than those. It might still have bugs and improvements to do on the device, and I believe we can do all those all together, so thats why I decided to start an opensource project.

I will post results as soon as I get them.

**Comment by peci1** [comment]
Hi, I'm afraid that benchmarking IMU performance on EKF results is insufficient. You need an independent positional reference. You should either mount a camera/lidar on the robot and run SLAM, or you should e.g. watch the robot with a ceiling-mounted camera and put an apriltag on top of the robot. Or, of course, you could use a Vicon-like system, but I'm pretty sure you don't have one at home :)

If I get it correctly, the imu_utils package only assesses the stationary IMU performance. But it's also important to know how it behaves in motion.

**Comment by altineller** [comment]
@peci1 I agree that benchmarking imu performance on ekf results nor measuring bias noise on stationary state. currently I dont have a lidar, but i will have in one or two months.

I could try the apriltag scheme, or aruco markers with my current tools.

I will keep this issue open and write when I progress.

**Comment by peci1** [comment]
Thanks!
